lib/zlib.js

```diff
@@ -146,7 +146,14 @@ function zlibOnError(message, errno) {
   var error = new Error(message);
   error.errno = errno;
   error.code = codes[errno];
-  this.emit('error', error);
+
+  if (this._initSuccess) {
+    this.emit('error', error);
+  } else {
+    process.nextTick(() => {
+      this.emit('error', error);
+    });
+  }
```

Comment:
That'll make it impossible to catch this error since there's on object returned to userland code yet to listen to the `error` event on. That's basically the reason for all this machinery, otherwise @cjihrig's one-line fix would be just fine.